### ISSUE TITLE
 Cannot read property 'getStackAddendumByID' of undefined

### ISSUE THREAD

**Issue by TDiDa** [alibaba/BizCharts#67]
BizCharts ^3.1.0-beta.1
react ^16.2.0

`
BizCharts.js:58909 Uncaught TypeError: Cannot read property 'getStackAddendumByID' of undefined
    at BizCharts.js:58909
    at Object.<anonymous> (BizCharts.js:71174)
    at t (BizCharts.js:30)
    at Object.<anonymous> (BizCharts.js:71221)
    at t (BizCharts.js:30)
    at Object.<anonymous> (BizCharts.js:51366)
    at t (BizCharts.js:30)
    at Object.<anonymous> (BizCharts.js:51884)
    at t (BizCharts.js:30)
    at Object.<anonymous> (BizCharts.js:418)
`

**Comment by weepy3641** [comment]
that's be fixed at 3.1.0-beta.2

**Comment by wangjianio** [comment]
But I also have this problem in 3.1.0-beta.2 or 3.1.0-beta.4
  "dependencies": {
    "@antv/data-set": "^0.8.2",
    "antd": "^3.0.3",
    "bizcharts": "^3.1.0-beta.2",
    "copy-to-clipboard": "^3.0.8",
    "jquery": "^3.2.1",
    "jshashes": "^1.0.7",
    "mockjs": "^1.0.1-beta3",
    "primer-markdown": "^3.7.3",
    "react": "^16.1.1",
    "react-app-rewire-less": "^2.1.0",
    "react-dom": "^16.1.1",
    "react-icons": "^2.2.7",
    "react-redux": "^5.0.6",
    "react-router": "^4.2.0",
    "react-router-dom": "^4.2.2",
    "react-scripts": "1.0.17",
    "redux": "^3.7.2"
  },

**Comment by dickeylth** [comment]
+1

**Comment by weepy3641** [comment]
@wangjianio @dickeylth 

https://stackoverflow.com/questions/46693961/locally-developed-react-component-imported-causes-error-on-build
https://github.com/facebookincubator/create-react-app/issues/3312

it seems that you have been used two version of react / react-dom.
try to clear the node-modules and install again？

**Comment by dickeylth** [comment]
@weepy3641 I've also found the link, and I reviewed my code and found noting about two versions of react/react-dom. I'm using [ant-design-pro](https://github.com/ant-design/ant-design-pro) with bizcharts.
Besides, when I use `roadhog build` and generated production mode bundle, the error occurred, but when I use `roadhog build --debug` and generated development mode bundle, the error wouldn't show. And that makes it hard to debug.

**Comment by weepy3641** [comment]
@dickeylth  
Can you show me your package.json ?
And what's your bizcharts version ?

**Comment by nikogu** [comment]
@dickeylth 

node version
npm version

**Comment by weepy3641** [comment]
@dickeylth 
 npm ls react
npm ls react-dom

**Comment by dickeylth** [comment]
> package.json

```json
{
  "name": "jellyfish",
  "version": "0.1.0",
  "description": "An out-of-box UI solution for enterprise applications",
  "private": true,
  "scripts": {
    "precommit": "npm run lint-staged",
    "start": "PORT=9001 roadhog server",
    "start:no-proxy": "cross-env NO_PROXY=true PORT=9001 roadhog server",
    "build": "roadhog build",
    "site": "roadhog-api-doc static && gh-pages -d dist",
    "analyze": "roadhog build --analyze",
    "lint:style": "stylelint \"src/**/*.less\" --syntax less",
    "lint": "eslint --ext .js src mock tests && npm run lint:style",
    "lint:fix": "eslint --fix --ext .js src mock tests && npm run lint:style",
    "lint-staged": "lint-staged",
    "lint-staged:js": "eslint --ext .js",
    "test": "jest",
    "test:all": "node ./tests/run-tests.js"
  },
  "dependencies": {
    "antd": "^3.0.0",
    "babel-polyfill": "^6.26.0",
    "babel-runtime": "^6.9.2",
    "bizcharts": "^3.1.0-beta.2",
    "classnames": "^2.2.5",
    "core-js": "^2.5.1",
    "dva": "^2.1.0",
    "g-cloud": "^1.0.2-beta",
    "g2": "^2.3.13",
    "g2-plugin-slider": "^1.2.1",
    "js-cookie": "^2.2.0",
    "lodash": "^4.17.4",
    "lodash-decorators": "^4.4.1",
    "lodash.clonedeep": "^4.5.0",
    "lodash.debounce": "^4.0.8",
    "moment": "^2.19.1",
    "numeral": "^2.0.6",
    "prop-types": "^15.5.10",
    "qs": "^6.5.0",
    "react": "^16.0.0",
    "react-container-query": "^0.9.1",
    "react-document-title": "^2.0.3",
    "react-dom": "^16.0.0",
    "react-fittext": "^1.0.0"
  },
  "devDependencies": {
    "babel-eslint": "^8.0.1",
    "babel-jest": "^21.0.0",
    "babel-plugin-dva-hmr": "^0.3.2",
    "babel-plugin-import": "^1.2.1",
    "babel-plugin-transform-class-properties": "^6.24.1",
    "babel-plugin-transform-decorators-legacy": "^1.3.4",
    "babel-plugin-transform-runtime": "^6.9.0",
    "babel-preset-env": "^1.6.1",
    "babel-preset-react": "^6.24.1",
    "cross-env": "^5.1.1",
    "cross-port-killer": "^1.0.1",
    "enzyme": "^3.1.0",
    "enzyme-adapter-react-16": "^1.0.2",
    "eslint": "^4.8.0",
    "eslint-config-airbnb": "^16.0.0",
    "eslint-plugin-babel": "^4.0.0",
    "eslint-plugin-compat": "^2.1.0",
    "eslint-plugin-import": "^2.2.0",
    "eslint-plugin-jsx-a11y": "^6.0.0",
    "eslint-plugin-markdown": "^1.0.0-beta.6",
    "eslint-plugin-react": "^7.0.1",
    "gh-pages": "^1.0.0",
    "jest": "^21.0.1",
    "lint-staged": "^4.3.0",
    "mockjs": "^1.0.1-beta3",
    "pro-download": "^1.0.0",
    "react-test-renderer": "^16.0.0",
    "redbox-react": "^1.3.2",
    "roadhog": "^1.3.1",
    "roadhog-api-doc": "^0.3.3",
    "rollbar": "^2.3.1",
    "stylelint": "^8.1.0",
    "stylelint-config-standard": "^17.0.0"
  },
  "optionalDependencies": {
    "nightmare": "^2.10.0"
  },
  "babel": {
    "presets": [
      "env",
      "react"
    ],
    "plugins": [
      "transform-decorators-legacy",
      "transform-class-properties"
    ]
  },
  "jest": {
    "setupFiles": [
      "<rootDir>/tests/setupTests.js"
    ],
    "testMatch": [
      "**/?(*.)(spec|test|e2e).js?(x)"
    ],
    "setupTestFrameworkScriptFile": "<rootDir>/tests/jasmine.js",
    "moduleFileExtensions": [
      "js",
      "jsx"
    ],
    "moduleNameMapper": {
      "\\.(css|less)$": "<rootDir>/tests/styleMock.js"
    }
  },
  "lint-staged": {
    "**/*.{js,jsx}": "lint-staged:js",
    "**/*.less": "stylelint --syntax less"
  },
  "browserslist": [
    "> 1%",
    "last 2 versions",
    "not ie <= 10"
  ]
}
```

```
~/dev/gitlab/jellyfish/site(feature/20171205_1584887_init_1 ✗) ll node_modules/react
lrwxr-xr-x  1 dickeylth  staff    19B 12  8 11:08 node_modules/react -> _react@16.2.0@react
~/dev/gitlab/jellyfish/site(feature/20171205_1584887_init_1 ✗) ll node_modules/react-dom
lrwxr-xr-x  1 dickeylth  staff    27B 12  8 11:08 node_modules/react-dom -> _react-dom@16.2.0@react-dom
~/dev/gitlab/jellyfish/site(feature/20171205_1584887_init_1 ✗) tnpm ls react
jellyfish@0.1.0 /Users/dickeylth/dev/gitlab/jellyfish/site
└── (empty)

[tnpm] subprocess:/usr/local/lib/node_modules/tnpm/node_modules/npm/bin/npm-cli.js exit code: 1, signal: null
[tnpm] mainprocess exit code: 1
~/dev/gitlab/jellyfish/site(feature/20171205_1584887_init_1 ✗) tnpm ls react-dom
tnpm -vjellyfish@0.1.0 /Users/dickeylth/dev/gitlab/jellyfish/site
└── (empty)

[tnpm] subprocess:/usr/local/lib/node_modules/tnpm/node_modules/npm/bin/npm-cli.js exit code: 1, signal: null
[tnpm] mainprocess exit code: 1
~/dev/gitlab/jellyfish/site(feature/20171205_1584887_init_1 ✗) tnpm -v
tnpm@5.9.0 (/usr/local/lib/node_modules/tnpm/bin/tnpm.js)
npm@5.6.0 (/usr/local/lib/node_modules/tnpm/node_modules/npm/lib/npm.js)
node@8.9.3 (/usr/local/bin/node)
npminstall@3.3.0 (/usr/local/lib/node_modules/tnpm/node_modules/npminstall/lib/index.js)
prefix=/usr/local (load prefix from: $ npm config get prefix)
darwin x64 17.3.0
registry=http://registry.npm.alibaba-inc.com
```

**Comment by weepy3641** [comment]
@TDiDa  @wangjianio  
fixed at bizcharts@3.1.0-beta.6
